### PR TITLE
fix(es/minifier): Fix analysis of var declaration after usage

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

`var_initialized` should be `true` even if the declaration of variable comes after its usage.

**Related issue:**

 - Closes https://github.com/swc-project/swc/issues/6039.

### CODE REVIEW

File: crates/swc_ecma_minifier/src/util/base54.rs [hunk 1]
```diff
@@ -186,16 +186,16 @@ impl CharFreq {
...
-        {
-            let considered = s
-                .chars()
-                .filter(|&c| Ident::is_valid_continue(c))
-                .collect::<String>();
-            if !considered.is_empty() {
-                tracing::debug!("Scanning: `{}` with delta {}", considered, delta);
-            }
-        }
+        // #[cfg(feature = "debug")]
```
Comment (kdy1) [inline crates/swc_ecma_minifier/src/util/base54.rs:189]: I commented this out as this disturbs debugging

File: crates/swc_ecma_minifier/src/pass/precompress.rs [hunk 2]
```diff
@@ -34,6 +34,23 @@ impl VisitMut for PrecompressOptimizer {
         }
     }
 
+    fn visit_mut_pat_or_expr(&mut self, n: &mut PatOrExpr) {
```
Comment (kdy1) [inline crates/swc_ecma_minifier/src/pass/precompress.rs:37]: I added a small normalization to reduce possibility of bugs

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

**Review by jridgewell — APPROVED** [review]
I'm confused by you're wording in the OP. Everything appears to be declared before use, when does an item get inserted into vars?

**Comment by kdy1** [comment]
I meant

```
function foo() {
    console.log(a)
}

var a = 1
```

`before` here means the order of visit

**Comment by kdy1** [comment]
Anyway, thank you!

**Pull request merged**